repeat "~" command failing

New project root folder.
Create HelloApp.scala file: 
object HelloApp extends App {
	println("Hello 1.0")
} 
set sbt.version=1.0.0-RC3 in .../project/build.properties
run sbt then "~ compile" or "~ run"

gives:

```scala
[error] java.lang.AssertionError: assertion failed                                                                                                                                                                            
[error]         at scala.Predef$.assert(Predef.scala:204)                                                                                                                                                                     
[error]         at sbt.internal.io.WatchState$.empty(SourceModificationWatch.scala:166)                                                                                                                                       
[error]         at sbt.Watched$.$anonfun$executeContinuously$1(Watched.scala:68)                                                                                                                                              
[error]         at scala.Option.getOrElse(Option.scala:121)                                                                                                                                                                   
[error]         at sbt.Watched$.executeContinuously(Watched.scala:68)                                                                                                                                                         
[error]         at sbt.BasicCommands$.$anonfun$continuous$2(BasicCommands.scala:215)                                                                                                                                          
[error]         at sbt.CommandUtil$.withAttribute(CommandUtil.scala:42)                                                                                                                                                       
[error]         at sbt.BasicCommands$.$anonfun$continuous$1(BasicCommands.scala:213)                                                                                                                                          
[error]         at sbt.Command$.$anonfun$applyEffect$4(Command.scala:125)                                                                                                                                                     
[error]         at sbt.Command$.$anonfun$applyEffect$2(Command.scala:121)                                                                                                                                                     
[error]         at sbt.MainLoop$.processCommand(MainLoop.scala:121)                                                                                                                                                           
[error]         at sbt.MainLoop$.$anonfun$next$2(MainLoop.scala:105)                                                                                                                                                          
[error]         at sbt.State$$anon$1.runCmd$1(State.scala:228)                                                                                                                                                                
[error]         at sbt.State$$anon$1.process(State.scala:234)                                                                                                                                                                 
[error]         at sbt.MainLoop$.$anonfun$next$1(MainLoop.scala:105)                                                                                                                                                          
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
[error]         at sbt.MainLoop$.next(MainLoop.scala:105)
[error]         at sbt.MainLoop$.run(MainLoop.scala:98)
[error]         at sbt.MainLoop$.$anonfun$runWithNewLog$1(MainLoop.scala:76)
[error]         at sbt.io.Using.apply(Using.scala:23)
[error]         at sbt.MainLoop$.runWithNewLog(MainLoop.scala:70)
[error]         at sbt.MainLoop$.runAndClearLast(MainLoop.scala:52)
[error]         at sbt.MainLoop$.runLoggedLoop(MainLoop.scala:37)
[error]         at sbt.MainLoop$.runLogged(MainLoop.scala:29)
[error]         at sbt.StandardMain$.runManaged(Main.scala:104)
[error]         at sbt.xMain.run(Main.scala:71)
[error]         at xsbt.boot.Launch$$anonfun$run$1.apply(Launch.scala:109)
[error]         at xsbt.boot.Launch$.withContextLoader(Launch.scala:128)
[error]         at xsbt.boot.Launch$.run(Launch.scala:109)
[error]         at xsbt.boot.Launch$$anonfun$apply$1.apply(Launch.scala:35)
[error]         at xsbt.boot.Launch$.launch(Launch.scala:117)
[error]         at xsbt.boot.Launch$.apply(Launch.scala:18)
[error]         at xsbt.boot.Boot$.runImpl(Boot.scala:41)
[error]         at xsbt.boot.Boot$.main(Boot.scala:17)
[error]         at xsbt.boot.Boot.main(Boot.scala)
[error] java.lang.AssertionError: assertion failed
[error] Use 'last' for the full log.
```

The commands work as expected (recompiling on saved file changes) in 0.13.16. This is running on Kubuntu 16.04, but yields similar output under windows 7.

sbt. version=1.0.0-RC3

--- a/main/src/main/scala/sbt/Defaults.scala
+++ b/main/src/main/scala/sbt/Defaults.scala
@@ -316,7 +316,9 @@ object Defaults extends BuildCommon {
                                      includeFilter in unmanagedSources,
                                      excludeFilter in unmanagedSources).value,
     watchSources in ConfigGlobal ++= {
-      val bases = unmanagedSourceDirectories.value
+      val baseDir = baseDirectory.value
+      val bases = unmanagedSourceDirectories.value ++ (if (sourcesInBase.value) Seq(baseDir)
+                                                       else Seq.empty)
       val include = (includeFilter in unmanagedSources).value
       val exclude = (excludeFilter in unmanagedSources).value
       bases.map(b => new Source(b, include, exclude))

